src/pytorch_lightning/trainer/connectors/callback_connector.py

```diff
@@ -95,16 +96,19 @@ def on_trainer_init(
     def _configure_accumulated_gradients(
         self, accumulate_grad_batches: Optional[Union[int, Dict[int, int]]] = None
     ) -> None:
-        grad_accum_callback = [cb for cb in self.trainer.callbacks if isinstance(cb, GradientAccumulationScheduler)]
+        grad_accum_callbacks: List[Callback] = [
+            cb for cb in self.trainer.callbacks if isinstance(cb, GradientAccumulationScheduler)
+        ]
 
-        if grad_accum_callback:
+        if grad_accum_callbacks:
+            grad_accum_callback: Callback
```

Comment:
Very small nit: In previous comment, I meant to move the typing even outside if :D

```suggestion
        grad_accum_callback: Callback
        if grad_accum_callbacks:
```